securedrop/client.py

```diff
@@ -37,31 +40,86 @@ def make_dict(self):
         }
 
 
+class AESWrapper(object):
+
+    def __init__(self, key):
+        self.bs = AES.block_size
+        self.key = hashlib.sha256(key.encode()).digest()
+
+    def encrypt(self, raw):
+        verify = hashlib.sha256((raw.encode())).hexdigest()
+        raw = self._pad(raw)
+        iv = Random.new().read(AES.block_size)
+        cipher = AES.new(self.key, AES.MODE_CBC, iv)
+        bites = base64.b64encode(iv + cipher.encrypt(raw.encode()))
+        return dict({"data": bites.decode('ascii'), "verify": verify})
+
+    def decrypt(self, enc):
+        data = enc["data"]
+        verify = enc["verify"]
+        data = base64.b64decode(data)
+        iv = data[:AES.block_size]
+        cipher = AES.new(self.key, AES.MODE_CBC, iv)
+        result = self._unpad(cipher.decrypt(data[AES.block_size:])).decode('utf-8')
+        if verify == hashlib.sha256((result.encode())).hexdigest():
+            return result
+        else:
+            raise RuntimeError("Decryption was not successful, could not verify input")
+
+    def _pad(self, s):
+        return s + (self.bs - len(s) % self.bs) * chr(self.bs - len(s) % self.bs)
+
+    @staticmethod
```

Comment:
nice, i didn't know about this annotation

Reply:
Same, it's a good thing somebody on stack overflow did.